platforms/software/dependency-management/src/main/java/org/gradle/internal/component/model/DefaultComponentGraphResolveState.java

```diff
@@ -41,8 +41,9 @@
 /**
  * Holds the resolution state for an external component.
  */
-public class DefaultComponentGraphResolveState<T extends ComponentGraphResolveMetadata, S extends ExternalComponentResolveMetadata> extends AbstractComponentGraphResolveState<T, S> {
+public class DefaultComponentGraphResolveState<T extends ComponentGraphResolveMetadata, S extends ExternalComponentResolveMetadata> extends AbstractComponentGraphResolveState<T> {
```

Comment:
🤔 Could we use more descriptive Generic type parameter names here?  `GRAPH_METADATA`/`ARTIFACT_METADATA`?  Or even just `GM`/`AM` instead of `T` and `S`, which seem arbitrary.

Reply:
I am personally not a fan of multi-character generics (or generics in general for that matter). 

I've changed them to `G` and `A`